devito/core/autotuning.py

```diff
@@ -56,9 +57,8 @@ def autotune(operator, arguments, tunable):
     # ... Defaults (basic mode)
     blocksizes = [OrderedDict([(i, v) for i in mapper]) for v in options['at_blocksize']]
     # ... Always try the entire iteration space (degenerate block)
-    datashape = [at_arguments[mapper[i].original_dim.symbolic_end.name] -
-                 at_arguments[mapper[i].original_dim.symbolic_start.name] for i in mapper]
-    blocksizes.append(OrderedDict([(i, mapper[i].iteration.extent(0, j))
+    datashape = [mapper[i].original_dim.symbolic_size.subs(arguments) for i in mapper]
```

Comment:
So this should probably be `extent`, I guess.  And then I can remove the extra declaration in `SubDimension`.